### PR TITLE
Increased possible pulsein length to ~65 ms.

### PR DESCRIPTION
**Pull request by DavePutz**
Issue #4659. Increased the possible max value of a pulsein pulse to 32767 us.  This is as high as it can go with affecting the interface code in shared-bindings/pulseio/PulseIn.c. Tested OK with a DHT11 temperature sensor and an infrared receiver.

### CODE REVIEW

**Comment by tannewt** [comment]
Isn't the existing shared-bindings limit 65535 because of the 16 bit unsigned int? Want to increase it further?

**Comment by DavePutz** [comment]
Yes, the limit can be raised to 65535 with the existing API. I'm not sure how long it would be reasonable to make it, though. The two devices I used to test (a DHT11 and an IR receiver) both were fine with a max of 4000 us. I'm not sure what devices might need pulses longer than 32 ms. Also, should an error be raised if the limit is exceeded? The current implementations on boards like atmel-samd seem to silently ignore pulses that are too long.

**Comment by tannewt** [comment]
My intention was that the pulse length would saturate at 65535 so it would represent anything longer than that.

File: ports/raspberrypi/common-hal/pulseio/PulseIn.c [hunk 1]
```diff
@@ -140,14 +140,19 @@ void common_hal_pulseio_pulsein_interrupt() {
...
             level_count = 1;
-            // ignore pulses that are too long and too short
-            if (result < MAX_PULSE && result > MIN_PULSE) {
-                self->buffer[buf_index] = result;
+	    // Pulses that are londger than MAX_PULSE will return MAX_PULSE
+	    if (result > MAX_PULSE ) {
+	        result = MAX_PULSE;
+	    }
+            // ignore pulses that are too short
+            if (result <= MAX_PULSE && result > MIN_PULSE) {
```
Comment (eigenein) [inline ports/raspberrypi/common-hal/pulseio/PulseIn.c:148]: I wonder, why there're hard-coded boundaries in the first place? As for the `MAX_PULSE`, my wild guess is that there're some dependencies on the buffer type which is `uint16_t[]`, is that correct? But for the `MIN_PULSE`, why would that be ever needed?

And I think, this should be stated in the API docs, may save a lot of time for someone.

**Comment by DavePutz** [comment]
The reason for a minimum pulse time was to ignore 'blips', which seem to be
especially common with the
infrared remote receiver. The maximum pulse time is mostly to give some
idea of when a series of pulses
has ended - otherwise we might sit for long periods of time waiting for a
high pulse to go low when a device
has, in fact, stopped sending pulses. I would prefer a timeout value to
handle that; but there is not currently
a timeout implemented in the PulseIn API.

**Review by tannewt — APPROVED** [review]
Thanks!

**Pull request merged**